### PR TITLE
lib: add internationalization effect

### PR DESCRIPTION
**Pull request by lisaw0**
Add internationalization effect to format date based on location.

- [x] I have read and accept the [Tokiwa Software Fuzion Contributor Agreement](https://github.com/tokiwa-software/fuzion/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

### CODE REVIEW

File: lib/internationalization/ISO_handler.fz [hunk 1]
```diff
@@ -0,0 +1,2 @@
+public ISO_handler : int_handler is
```
Comment (fridis) [inline lib/internationalization/ISO_handler.fz]: please add the header to every file. 

also, add a small comment explaining what this file does.

File: lib/internationalization/ISO_handler.fz [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature internationalization.ISO_handler
+#
+# -----------------------------------------------------------------------
+
+# ISO_handler -- provides data for ISO 8601 format
+#
+public ISO_handler : int_handler is
+    redef date(d i32,m i32,y i32) => "{y}-{m}-{d}"
```
Comment (fridis) [inline lib/internationalization/ISO_handler.fz:27]: please keep one empty line before and after feature declarations and add a comment for each feature, e.g., 

```suggestion
# ISO_handler -- provides data for ISO 8601 format
#
public ISO_handler : int_handler is

    # format date using ISO formatting rules, see https://en.wikipedia.org/wiki/ISO_8601
    # 
    # ex. `date 2024 4 12` results in `2024-04-12`.
    #
    redef date(d i32,m i32,y i32) 
    => 
      "{y}-{m}-{d}"
```

Comment (fridis) [inline lib/internationalization/ISO_handler.fz:27]: The implementation is wrong, there are no leading `0`!

**Review by fridis — CHANGES_REQUESTED** [review]

File: lib/internationalization/german_handler.fz [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#
+# -----------------------------------------------------------------------
+
+# german_handler -- provides data for german format
+#
+public german_handler : int_handler is
+
+    #format date using the most common german formatting rules, see https://en.wikipedia.org/wiki/Date_and_time_notation_in_Europe#Germany
+    #
+    # ex.`date 2024 4 12` results in `4.12.2024`.
```
Comment (fridis) [inline lib/internationalization/german_handler.fz:30]: The example is not correct, should be:
```suggestion
    # ex.`date 2024 4 12` results in `12.4.2024`.
```

File: lib/internationalization/american_handler.fz [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+#
+public american_handler : int_handler is
+
+    #format date using the most common american formatting rules, see https://en.wikipedia.org/wiki/Date_and_time_notation_in_the_United_States
+    #
+    # ex.`date 2024 4 12` results in `12/04/2024`.
+    #
+    redef date(d i32,m i32,y i32)
+    =>
+        "{($m).pad_left("0", 2)}/{($d).pad_left("0", 2)}/{y}"
```
Comment (fridis) [inline lib/internationalization/american_handler.fz:34]: In Fuzion, the preferred syntax for a call is `f a b`, not `f(a, b)`:
```suggestion
        "{($m).pad_left "0" 2}/{($d).pad_left "0" 2}/{y}"
```

File: lib/internationalization/ISO_handler.fz [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+#
+#  Source code of Fuzion standard library feature internationalization.ISO_handler
+#
+# -----------------------------------------------------------------------
+
+# ISO_handler -- provides data for ISO 8601 format
+#
+public ISO_handler : int_handler is
+
+    #format date using ISO formatting rules, see https://en.wikipedia.org/wiki/ISO_8601
```
Comment (fridis) [inline lib/internationalization/ISO_handler.fz:28]:
```suggestion
    # format date using ISO formatting rules, see https://en.wikipedia.org/wiki/ISO_8601
```

File: lib/internationalization/american_handler.fz [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+#
+#  Source code of Fuzion standard library feature internationalization.american_handler
+#
+# -----------------------------------------------------------------------
+
+# american_handler -- provides data for american format
+#
+public american_handler : int_handler is
+
+    #format date using the most common american formatting rules, see https://en.wikipedia.org/wiki/Date_and_time_notation_in_the_United_States
```
Comment (fridis) [inline lib/internationalization/american_handler.fz:28]:
```suggestion
    # format date using the most common American formatting rules, see
    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_the_United_States
```

File: lib/internationalization/german_handler.fz [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+#
+#  Source code of Fuzion standard library feature internationalization.german_handler
+#
+# -----------------------------------------------------------------------
+
+# german_handler -- provides data for german format
+#
+public german_handler : int_handler is
+
+    #format date using the most common german formatting rules, see https://en.wikipedia.org/wiki/Date_and_time_notation_in_Europe#Germany
```
Comment (fridis) [inline lib/internationalization/german_handler.fz:28]:
```suggestion
    # format date using the most common German formatting rules, see 
    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_Europe#Germany
```

File: lib/internationalization/japanese_handler.fz [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+#
+#  Source code of Fuzion standard library feature internationalization.japanese_handler
+#
+# -----------------------------------------------------------------------
+
+# japanese_handler -- provides data for japanese format
+#
+public japanese_handler : int_handler is
+
+    #format date using the most common japanese formatting rules, see https://en.wikipedia.org/wiki/Date_and_time_notation_in_Japan
```
Comment (fridis) [inline lib/internationalization/japanese_handler.fz:28]:
```suggestion
    # format date using the most common japanese formatting rules, see
    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_Japan
```

File: lib/internationalization/korean_handler.fz [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+#
+#  Source code of Fuzion standard library feature internationalization.korean_handler
+#
+# -----------------------------------------------------------------------
+
+# korean_handler -- provides data for korean format
+#
+public korean_handler : int_handler is
+
+    #format date using the most common korean formatting rules, see https://en.wikipedia.org/wiki/Date_and_time_notation_in_South_Korea
```
Comment (fridis) [inline lib/internationalization/korean_handler.fz:28]:
```suggestion
    # format date using the most common korean formatting rules, see
    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_South_Korea
```

File: lib/internationalization.fz [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature internationalization
+#
+# -----------------------------------------------------------------------
+
+# internationalization -- unit feature to group features that internationalize certain facts provided
```
Comment (fridis) [inline lib/internationalization.fz:24]: not really a sentence. how about
```suggestion
# internationalization -- unit feature to group features related to specific 
# geographical, political, or cultural regions such formatting of dates, 
# numbers, currencies, etc.
```

File: lib/internationalization/int_eff.fz [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature internationalization.int_eff
+#
+# -----------------------------------------------------------------------
+
+# int_eff -- effect that provides data for internationalization
```
Comment (fridis) [inline lib/internationalization/int_eff.fz:24]:
```suggestion
# int_eff -- effect that provides operations that are specific for a geographical, political,
# or cultural region. Such operations include formatting of dates,  numbers, currencies,
# etc.
```

File: lib/internationalization/int_handler.fz [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature internationalization.int_handler
+#
+# -----------------------------------------------------------------------
+
+# int_handler -- provides abstract features for internationalization
```
Comment (fridis) [inline lib/internationalization/int_handler.fz:24]:
```suggestion
# int_handler -- abstract handler to be used by `int_eff`
```

File: lib/internationalization/ISO_handler.fz [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    #
+    # ex.`date 2024 4 12` results in `2024-04-12`.
+    #
+    redef date(d i32,m i32,y i32)
```
Comment (fridis) [inline lib/internationalization/ISO_handler.fz:32]: Please add spaces after the commas, also at all other locations were feature argument lists are declared.
```suggestion
    redef date(d i32, m i32, y i32)
```

**Review by fridis — CHANGES_REQUESTED** [review]
Looks good. 

I have added a number of smaller suggestions, mostly affecting formatting or some text formulation. 

Please fix and then this should be ready to be merged!

File: lib/internationalization/korean_handler.fz (hunk 9 above)
Comment (lisaw0) [inline lib/internationalization/korean_handler.fz:28]: # format date using the most common Korean formatting rules, see
	    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_South_Korea

File: lib/internationalization/japanese_handler.fz (hunk 8 above)
Comment (lisaw0) [inline lib/internationalization/japanese_handler.fz:28]:
```suggestion
    # format date using the most common Japanese formatting rules, see
	    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_Japan
```

File: lib/internationalization/korean_handler.fz [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+#  Source code of Fuzion standard library feature internationalization.korean_handler
+#
+# -----------------------------------------------------------------------
+
+# korean_handler -- provides data for korean format
+#
+public korean_handler : int_handler is
+
+    # format date using the most common Korean formatting rules, see
+	    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_South_Korea
```
Comment (michaellilltokiwa) [inline lib/internationalization/korean_handler.fz:29]:
```suggestion
    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_South_Korea
```

File: lib/internationalization/japanese_handler.fz [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+#  Source code of Fuzion standard library feature internationalization.japanese_handler
+#
+# -----------------------------------------------------------------------
+
+# japanese_handler -- provides data for japanese format
+#
+public japanese_handler : int_handler is
+
+    # format date using the most common Japanese formatting rules, see
+	    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_Japan
```
Comment (michaellilltokiwa) [inline lib/internationalization/japanese_handler.fz:29]:
```suggestion
    # https://en.wikipedia.org/wiki/Date_and_time_notation_in_Japan
```

File: lib/internationalization.fz [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature internationalization
+#
+# -----------------------------------------------------------------------
+
+# internationalization -- unit feature to group features related to specific 
+# geographical, political, or cultural regions such formatting of dates,
```
Comment (michaellilltokiwa) [inline lib/internationalization.fz:25]:
```suggestion
# geographical, political, or cultural regions such as formatting of dates, 
```

**Review by michaellilltokiwa — APPROVED** [review]

**Review by fridis — APPROVED** [review]

**Pull request merged**